### PR TITLE
Guide: How to Create Your First Mock Server

### PR DESCRIPTION
**Pull request by Aman333Saxena**
A blog for the "How-To Guides" category, about How to Create Your First Mock Server simulating APIs. It consists of all the extensive details, step-by-step instructions with images and resources links.

Update1: Author Bio and the Headshot Image

### CODE REVIEW

**Comment by hectorkambow** [comment]
@Aman333Saxena if you could please make sure you have updated your repo to the most recent - and/or follow the instructions to upload your article and image file - they do not seem to be in the correct place.
images don't need to be in a separate folder

**Comment by Aman333Saxena** [comment]
Hello, @hectorkambow 
I have added all the images and the article in the same folder as you have asked.  The current Repo also looks clean.  
I have updated image references as well. All good to go :)

**Comment by hectorkambow** [comment]
Good afternoon @manibatra  - could you please give this article a tech review when you get a chance. Thank you 🚀

File: articles/guide-to-create-mock-server/index.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+###  Why are Mock Servers so useful?
+- Mock Servers start working against a service API even before the service is available. If an API or service is not yet fully developed they can mock the API allowing any team who is using the service to start work without being delayed.
+- They isolate development teams during the initial development phases when the APIs / services may be extremely unstable and volatile. Using MockServer allows development work to continue even when an external service fails.
+
+![Usage Image](/engineering-education/guide-to-create-mock-server/image01.png) *Fig 2: www.mock-server.com*
+
+- Mock Servers easily recreate all types of responses for HTTP dependencies such as REST or RPC services to test applications easily and effectively. They isolate the system-under-test to ensure tests run reliably and only fail when there is a genuine bug.
+- These servers easily setup mock responses independently for each test to ensure test data is encapsulated with each test. Avoid sharing data between tests that are difficult to manage and maintain this also decreases the risks of tests infecting each other.
```
Comment (manibatra) [inline articles/guide-to-create-mock-server/index.md:36]: `They avoid sharing data between tests that are difficult to manage and maintain. This also decreases the risks of tests infecting each other.`

File: articles/guide-to-create-mock-server/index.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+- Now, you will be able to see a new tab asking for some basic information about the server. Choose whether you want to mock a **new API** or an **existing collection**. If you create a new API to mock, you will select a request method and enter the request path, response code, and response body or else use an existing collection to mock.
+
+![Select Image](/engineering-education/guide-to-create-mock-server/image3.png)
+
+- When you have selected or created the request you want to mock, click **Next**.
+
+![Set up Image](/engineering-education/guide-to-create-mock-server/image4.png)
+
+- In the **Set up the mock server** tab, you can configure your mock server.
+1. Enter the name of the mock.
```
Comment (manibatra) [inline articles/guide-to-create-mock-server/index.md:64]: name of the mock server?

File: articles/guide-to-create-mock-server/index.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+4. Check the checkbox if you want to save the mock server URL as an environment variable.
+5. Click Next to continue.
+
+- In the **Next steps** tab, you will see a list of suggested next steps to maximize the effectiveness of your mock server.
+
+![Next Step Image](/engineering-education/guide-to-create-mock-server/image5.png)
+
+- If the setting process went well without an error, you'll be able to see your Mock Server in the left corner on the **Collections** panel.
+
+You can view and search the details of calls to your mock servers using the mock call log. Open a mock from the Postman app by clicking it in Collections, in APIs, or by switching to Browse > Mocks and clicking the mock name.
```
Comment (manibatra) [inline articles/guide-to-create-mock-server/index.md:76]: Open a mock server?

File: articles/guide-to-create-mock-server/index.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+#### Step 4: I am all done with my server. How do I create a Router Request?
+
+![Request Image](/engineering-education/guide-to-create-mock-server/image6.png)
+
+- In the left-hand corner, you will see **Setting Option(3 Dots)** in the Collections panel. On clicking that, you'll be able to see several options. Choose **Add Request**, by which a large number of Request Methods, (i.e. GET, POST) will be made available to you, choose whatever you need. Make sure you have an example saved for the request in the collection you have the mock connected to.
```
Comment (manibatra) [inline articles/guide-to-create-mock-server/index.md:82]: see the **Setting**

**Comment by Aman333Saxena** [comment]
@hectorkambow @manibatra 
I have fixed all the requested changes. It looks great after the edits. 
Thank you so much for an in-detail review. 🚀 ✨

**Pull request merged**